### DISCUSSION TITLE
Can't paint materials

### DISCUSSION THREAD

**Discussion by ac-arcana** [blackears/cyclopsLevelBuilder#144] [Q&A] [answered]
First time user here (used to use Qodot but 1.0.2 appears to have everything TrenchBroom does). I am having trouble with painting materials, I can't get a material to show up at all.

 I set up the latest version as an add on based on the 1.0.2 video. I have made some blocks, now I am trying to apply a material. 
I have the material added to the materials dock. 
I have selected a brush and double clicked the material, and I've also tried the material brush tool. 
When I select the block I can see the material has been put into the materials array. However, I still only see the checkered material. My view drop down is set to materials.

What could I be missing to get the material to show up?

![image](https://github.com/blackears/cyclopsLevelBuilder/assets/5461795/ac0dca73-9da0-47a2-99be-0b103118eed3)

**Comment by blackears** [comment]
I'm not sure.  It looks like it should be fine.  Have you tried this with the latest development version of Cyclops?  Could you send me a file demonstrating this?

**Comment by ac-arcana** [comment]
I still haven't figured this out but I've continued building. The odd part is if I select the material before I make a brush, then the material shows up on the new brush. I can change the material after that as well. At least I have a work around now...

What files would you need? A scene file and associated materials?

I just downloaded the zip from the github today. Is there a branch I should use?

**Reply by blackears** [reply]
You can checkout the project.  There is a python script called create_addon_archine.py that you can run to create a new archive.  You can also just copy everything in the /addons directory into your project instead.

Just post a zip file containing the entire project that shows the faulty behavior so that I can run it.

**Comment by ac-arcana** [comment]
[Test_Cyclops_Material_Issue.zip](https://github.com/blackears/cyclopsLevelBuilder/files/14551751/Test_Cyclops_Material_Issue.zip)
Alright here is the minimum repro, I didn't want to give you a ton of bloat from other files.
I think this problem is occuring because I am making the brush before having any materials set up in Cyclops. Once there are materials set up in Cyclops new brushes behave normally.

**Comment by blackears** [comment, marked as answer]
It looks like some important lines got commented out.  I've restored them.  But part of the problem might also have been that you selected one of the vertex, edge or face brushes before clicking on the material and did not select any faces.

Please check out the latest dev version and let me know if this fixes things.

**Reply by ac-arcana** [reply]
Wow, thank you for looking at this so quickly! I'll pay attention for possibly selecting things like that in the future. I'm really loving what I can do with Cyclops.

Do you plan to add a texture window like in TrenchBroom? Where you can move the UI position and scale against the face you have selected? It's a minor thing but kind of nice. Regardless, this workflow is so much better than that of Qodot now. 1.0.2 seems like a huge upgrade. Thanks!

**Comment by blackears** [comment]
I am working on improvements to how materials work at the moment when I can find the free time. :)  I would like to improve how UV layout is handled, but it might be a while before I get to it.
